crates/sui-json-rpc-types/src/sui_object.rs

```diff
@@ -0,0 +1,452 @@
+use colored::Colorize;
+use schemars::JsonSchema;
+use serde::Deserialize;
+use serde::Serialize;
+use std::fmt;
+use std::fmt::Write;
+use std::fmt::{Display, Formatter};
+
+use sui_protocol_config::ProtocolConfig;
+use sui_types::base_types::{
+    ObjectDigest, ObjectID, ObjectRef, ObjectType, SequenceNumber, TransactionDigest,
+};
+use sui_types::error::{UserInputError, UserInputResult};
+use sui_types::gas_coin::GasCoin;
+use sui_types::move_package::MovePackage;
+use sui_types::object::{Data, MoveObject, Object, ObjectRead, Owner};
+use sui_types::parse_sui_struct_tag;
+
+use crate::SuiObjectRef;
+use crate::{
+    SuiData, SuiMoveObject, SuiMoveStruct, SuiMoveValue, SuiObject, SuiParsedData, SuiRawData,
+};
+
+#[derive(Serialize, Deserialize, Debug, JsonSchema, Clone)]
+#[serde(tag = "status", content = "details", rename = "ObjectRead")]
+pub enum SuiObjectWithStatus {
+    Exists(SuiObjectData),
+    NotExists(ObjectID),
+    Deleted(SuiObjectRef),
+}
+
+#[derive(Debug, Clone, Deserialize, Serialize, JsonSchema, Eq, PartialEq)]
+#[serde(rename_all = "camelCase", rename = "ObjectData")]
+pub struct SuiObjectData {
+    pub object_id: ObjectID,
+    /// Object version.
+    pub version: SequenceNumber,
+    /// Base64 string representing the object digest
+    pub digest: ObjectDigest,
+    /// The type of the object. Default to be Some(T) unless SuiObjectContentOptions.showType is set to False
+    #[serde(rename = "type", skip_serializing_if = "Option::is_none")]
+    pub type_: Option<String>,
+    // Default to be None because otherwise it will be repeated for the getObjectsOwnedByAddress endpoint
+    /// The owner of this object. Default to be None unless SuiObjectContentOptions.showOwner is set
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub owner: Option<Owner>,
+    /// The digest of the transaction that created or last mutated this object. Default to be None unless
+    /// SuiObjectContentOptions.showPreviousTransaction is set
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub previous_transaction: Option<TransactionDigest>,
+    /// The amount of SUI we would rebate if this object gets deleted.
+    /// This number is re-calculated each time the object is mutated based on
+    /// the present storage gas price.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub storage_rebate: Option<u64>,
+    // TODO: uncomment the following in the next PR
+    // /// The Display metadata for frontend UI rendering
+    // pub display: Option<BTreeMap<String, String>>,
+    /// Move object content or package content, default to be None unless SuiObjectContentOptions.showContent is set
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub content: Option<SuiParsedData>,
+    /// Move object content or package content in BCS, default to be None unless SuiObjectContentOptions.showContent is set
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub bcs: Option<SuiRawData>,
```

Comment:
so type_ is defaulted to Some but others is defaulted to None..... this is a bit confusing imo, user will have to always go back to the API reference to check the default..... shell we just default everything to None?